src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForFailedStatus.java

```diff
@@ -145,16 +150,94 @@ public void should_record_failure_of_upload_when_document_management_throws_exce
         EnvelopeState envelopeState = envelopeStates.get(0);
         assertThat(envelopeState)
             .extracting("container", "zipFileName", "status")
-            .hasSameElementsAs(ImmutableList.of(testContainer.getName(), ZIP_FILE_NAME, DOC_UPLOAD_FAILURE));
+            .hasSameElementsAs(ImmutableList.of(testContainer.getName(), ZIP_FILE_NAME_SUCCESS, DOC_UPLOAD_FAILURE));
         assertThat(envelopeState.getId()).isNotNull();
         assertThat(envelopeState.getReason()).isEqualTo(throwable.getMessage());
         assertThat(envelopeState.getEnvelope().getId()).isEqualTo(actualEnvelope.getId());
     }
 
-    private void uploadZipToBlobStore() throws Exception {
-        byte[] zipFile = toByteArray(getResource(ZIP_FILE_NAME));
+    @Test
+    public void should_record_generic_failure_when_zip_does_not_contain_metadata_json() throws Exception {
+        // given
+        String noMetafileZip = "2_24-06-2018-00-00-00.zip";
```

Comment:
does the file name need to be in this particular format?
ie, can the file be called something like `no-metafile.zip`?

Reply:
it is most likely taken from the documentation spec:

> Zip file format : unqiueId_datetimecreated.zip